custom/enikshay/tasks.py

```diff
@@ -193,6 +206,11 @@ def run_batch(self, case_ids):
 
         return BatchStatus(update_count, noupdate_count, success_count, errors, case_batches, t.interval)
 
+    @staticmethod
+    def _update_ledger_values(episode_case, adherence_date, adherence_source, adherence_value):
+        if adherence_date and adherence_source and adherence_value:
+            update_ledger_for_adherence(episode_case, adherence_date, adherence_source, adherence_value)
```

Comment:
currently, this will make a single form submission per episode case that needs to be updated, which was causing us issues before. this should be batched as we do with the case updates.

Reply:
@proteusvacuum what issues did you face earlier with single form/episode?

Reply:
we update ~20,000 episode cases a night and there is a time overhead in submitting a form, even if it only contains a single update. By batching the form submissions you can reduce the amount of time the task takes. If you don't batch them, I don't think this would complete in a reasonable amount of time.

Reply:
the one drawback i have seen to that is loading this form on case detail page. But i guess we ignored that considering the benefits of batching here.